### DISCUSSION TITLE
Nested navigation

### DISCUSSION THREAD

**Discussion by vhmvd** [android/nowinandroid#1701] [Q&A]
I have nested navigation in my multi module app. 

I can see, clear as day, one of the example provided [here](https://github.com/android/nowinandroid/blob/main/feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/navigation/ForYouNavigation.kt). This example works well where the app needs to provide the destination and it's nav behaviour, e.g. Routing to detail of something.

My module requires me to navigate as 

```
Location Screen > Navigate to children of selected location > [if has child] navigate more [else] show details
```

What would be the ideal way to achieve this behaviour?

So far I have avoided passing navController because from [docs](https://developer.android.com/develop/ui/compose/navigation):

> This means that you shouldn't pass the navController [directly into any composable](https://developer.android.com/guide/navigation/design) and instead pass navigation callbacks as parameters.

### TLDR

How to navigate and access `NavController` within `Game` if it's a separate module from example [here](https://developer.android.com/guide/navigation/design/nested-graphs#compose).

**Comment by vhmvd** [comment]
> What would be the ideal way to achieve this behaviour?

Can be done by the following code, or using a shared view model.

```kotlin
controller.navigate(Child(/* needed IDs here */))
```

> So far I have avoided passing navController because from [docs](https://developer.android.com/develop/ui/compose/navigation):

> How to navigate and access NavController within Game if it's a separate module from example [here](https://developer.android.com/guide/navigation/design/nested-graphs#compose).

The detail I missed or overlooked was that I can pass the `navController` to a graph builder and access it in that way.
